FEC-69: feature(ui components): added a new tree view multi-select component

MaxKalashnyk opened the pull request:
Added a new Tree view multi-select component. It could have these states (with children and without children options):
<img width="472" alt="Screenshot 2023-08-01 at 12 31 50" src="https://github.com/uptechteam/fe-cookbook/assets/13544983/63ab5593-3124-4217-a58e-1c98961fe3f1">

<img width="341" alt="Screenshot 2023-08-01 at 12 30 45" src="https://github.com/uptechteam/fe-cookbook/assets/13544983/96ebe74a-aab0-4e3c-b449-dee967afbc57">

yarikhot commented:
Are you sure we need to use another package instead of material?
Did you try to make three multi-select with the Material?

MaxKalashnyk commented:
> @MaxKalashnyk Are you sure we need to use another package instead of material? Did you try to make three multi-select with the Material?

Nope, I haven't tried to build this component with MUI. Of course, we can create this component with MUI but it'll take a lot of time to cover all of the use cases from examples on screenshots.

MaxKalashnyk commented:
@yarikhot, I've updated this component description and moved the demo to Codesandbox. 
I tried to change the class component to functional but this component didn't work properly, so I left the existing solution here for now.
